bootstrap/secret/secure.go

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright 2018 Dell Inc.
+ * Copyright 2020 Intel Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package secret
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/authtokenloader"
+	"github.com/edgexfoundry/go-mod-secrets/secrets"
+
+	"sync"
+	"time"
+)
+
+// SecureProvider implements the SecretProvider interface
+type SecureProvider struct {
+	secretClient  secrets.SecretClient
+	lc            logger.LoggingClient
+	loader        authtokenloader.AuthTokenLoader
+	configuration interfaces.Configuration
+	secretsCache  map[string]map[string]string // secret's path, key, value
+	cacheMutex    *sync.Mutex
+	lastUpdated   time.Time
+}
+
+// NewSecureProvider creates & initializes Provider instance for secure secrets.
+func NewSecureProvider(config interfaces.Configuration, lc logger.LoggingClient, loader authtokenloader.AuthTokenLoader) *SecureProvider {
+	provider := &SecureProvider{
+		configuration: config,
+		lc:            lc,
+		loader:        loader,
+		secretsCache:  make(map[string]map[string]string),
+		cacheMutex:    &sync.Mutex{},
```

Comment:
I thought RWMutex is a better option here, we don't need to lock for lots of concurrent GetSecrets calls
https://golang.org/pkg/sync/#RWMutex

We can use RLock and RUnlock for getSecretsCache